### PR TITLE
Add get by user support to audit service

### PR DESCRIPTION
**Pull request by HaniAlshikh**
let administrators or people like CISOs get detailed information about a user to answer questions like

- How did a user get that role?
- How did a user become a tenant member?
- How did a user get access to monoskope at all?

### CODE REVIEW

File: internal/eventstore/usecases/retrieve_events.go [hunk 1]
```diff
@@ -89,3 +111,14 @@ func (u *RetrieveEventsUseCase) Run(ctx context.Context) error {
 
 	return nil
 }
+
+func (u *RetrieveEventsUseCase) load(ctx context.Context, storeQueries []*es.StoreQuery) (es.EventStreamReceiver, error) {
+	switch u.gate {
```
Comment (jastBytes) [inline internal/eventstore/usecases/retrieve_events.go:116]: Why this gate? Couldn't you just make this decision based on if there is more than one filter?

**Review by jastBytes — CHANGES_REQUESTED** [review]

File: internal/eventstore/usecases/retrieve_events.go (hunk 1 above)
Comment (HaniAlshikh) [reply internal/eventstore/usecases/retrieve_events.go:116]: Yeah i wasn't sure about this one either but I thought maybe just maybe it made sense in the future to support more gates like and, xor, etc...., What do you think?

But yeah you are right I can for sure base the decision on the number filters.

Comment (jastBytes) [reply internal/eventstore/usecases/retrieve_events.go:116]: If we need something like this in the future we can still refactor it like that. It always makes sense to keep things simple and don't plan for use-cases that might never come. ;-)

Comment (HaniAlshikh) [reply internal/eventstore/usecases/retrieve_events.go]: Alrighty 👍

**Review by jastBytes — APPROVED** [review]

**Pull request merged**